### PR TITLE
Add playground module and refactor changes

### PR DESCRIPTION
**Pull request by kaizhangNV**
Add a playground module so that we can put some utility functions into this module.

### CODE REVIEW

**Comment by kaizhangNV** [comment]
Let's merge this change after demo.

File: compiler.js [hunk 1]
```diff
@@ -58,6 +68,25 @@ void printMain(uint3 dispatchThreadID : SV_DispatchThreadID)
...
+float4 imageMain(uint2 dispatchThreadID, int2 screenSize)
+{
+    return float4(0);
+}
+`;
+
+const emptyPrintShader = `
+import playground;
+
+int printMain(uint2 dispatchThreadID, int2 threadGroupSize)
```
Comment (csyonghe) [inline compiler.js:83]: printMain() does not need to have any parameters.

File: compiler.js [hunk 2]
```diff
@@ -369,19 +412,15 @@ class SlangCompiler
...
                 return null;
-            }
 
-            var components = new this.slangWasmModule.ComponentTypeList();
-            components.push_back(userModule);
+            if (!this.loadModule(slangSession, "user", shaderSource, components))
+                return null;
 
-            if (this.addActiveEntryPoints(slangSession, shaderSource, entryPointName, isWholeProgram, userModule, components) == false)
+            if (this.addActiveEntryPoints(slangSession, shaderSource, entryPointName, isWholeProgram, components.get(1), components) == false)
```
Comment (csyonghe) [inline compiler.js:423]: in the `finalize` block down below, also need to `components.delete()`.

File: compiler.js [hunk 3]
```diff
@@ -22,27 +22,37 @@ inline float encodeColor(float4 color)
 }
 
 [shader("compute")]
+[numthreads(16, 16, 1)]
```
[the remark this answers is not in the document]
Comment (kaizhangNV) [reply compiler.js:25]: like what?

[the remark this answers is not in the document]
Comment (csyonghe) [reply compiler.js:25]: Anything that is different so it is not confusing? Like imageEntryPoint

**Review by csyonghe — APPROVED** [review]

File: compiler.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (kaizhangNV) [reply compiler.js:25]: sure, will change in next PR.

[the remark this answers is not in the document]
Comment (kaizhangNV) [reply compiler.js:25]: wait, thinking about this again.
We will display the entry point name on the dropdown, will this cause more confusing that users have no idea what that is?

[the remark this answers is not in the document]
Comment (csyonghe) [reply compiler.js:25]: Hmm good point, let's keep it as is then.

**Pull request merged**